src/lib/ext-activitylog.js

```diff
@@ -3,26 +3,199 @@ import { save } from './save-load.js';
+          );
+      }
+    } else if (event.type === 'change') {
+      this.handleChangeOnCheckbox(event);
+    } else {
+      throw new Error(`wrong event type - ${event.type}`);
+    }
+  }
+
+  toggleFilterListDisplay(toggleBtn, checkboxList) {
+    checkboxList.hidden = !checkboxList.hidden;
+    this.toggleFilterBtnIcon(toggleBtn, checkboxList.hidden);
+  }
+
+  toggleFilterBtnIcon(element, isListHidden) {
+    isListHidden
+      ? element
+          .querySelector('.arrow-down')
+          .classList.replace('arrow-down', 'arrow-right')
+      : element
+          .querySelector('.arrow-right')
+          .classList.replace('arrow-right', 'arrow-down');
+  }
+
+  handleChangeOnCheckbox(event) {
+    let target = null;
+    switch (event.currentTarget) {
+      case this.extCheckboxList:
+        target = 'id';
+        break;
+      case this.viewTypeCheckboxList:
+        target = 'viewType';
+        break;
+      case this.apiTypeCheckboxList:
+        target = 'type';
+        break;
+      default:
+        break;
+    }
+
+    if (target) {
+      const filterObj = {
+        filterParam: { value: event.target.closest('input').value, target },
+        checked: event.target.closest('input').checked,
+      };
+
+      const filterEvent = new CustomEvent('filter', { detail: filterObj });
+      event.currentTarget.dispatchEvent(filterEvent);
+    }
+  }
+
+  handleNewLog(object) {
+    this.logView.addNewRow(object);
+    this.updateFilterCheckboxes(object.log);
+  }
+
+  updateFilterCheckboxes(log) {
+    for (const key of Object.keys(log)) {
+      switch (key) {
+        case 'id':
+          this.addNewCheckbox(this.extCheckboxList, log.id);
+          break;
+        case 'viewType':
+          this.addNewCheckbox(this.viewTypeCheckboxList, log.viewType);
+          break;
+        case 'type':
+          this.addNewCheckbox(this.apiTypeCheckboxList, log.type);
+          break;
+        default:
+          break;
+      }
+    }
   }
 
-  addTableRows(logs) {
-    this.logTable.addNewRows(logs);
+  addNewCheckbox(checkboxList, labelText) {
+    if (!this.isCheckboxExist(checkboxList, labelText)) {
+      const newCheckbox = this.checkboxTemplate.cloneNode(true);
+      newCheckbox.querySelector('label p').textContent = labelText;
+      newCheckbox.querySelector('input').value = labelText;
+      newCheckbox.querySelector('input').checked = true;
+      checkboxList.appendChild(newCheckbox);
+    }
+  }
+
+  isCheckboxExist(checkboxList, labelText) {
+    const checkboxLabels = checkboxList.querySelectorAll('label p');
+    for (const label of checkboxLabels) {
+      if (label.textContent === labelText) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  showRows({ filterParam, filterObject, isFilterMatchingFn }) {
+    const { target, value } = filterParam;
+
+    const hiddenRows = this.logView.logTableWrapper.querySelectorAll(
+      'tbody tr[hidden]'
```

Comment:
The `logTableWrapper` and its `tr` elements is an internal implementation detail of `logView`. You shouldn't access those from external classes, because that breaks encapsulation. Encapsulation exists to make it easier to reason about components, but if you break the encapsulation this benefit is lost.

In this file you're passing the filter function to the `logView`, twice.

Instead of doing that here, you could also pass the model (or at least the filter object) to the `logView`, and let `logView` re-apply the filter if needed (via events or explicit function calls).

Reply:
This has been done in logView component.